### DISCUSSION TITLE
Logging / Troubleshooting Linux auth failure

### DISCUSSION THREAD

**Discussion by downtownpartytime** [Netatalk/netatalk#1508] [Q&A] [answered]
I compiled and installed from netatalk-3.2.9.tar.xz on Debian 12.1 with meson and can see the share on my mac(with the name I set in afp.conf), but it does not accept the username/password (it was default using PAM, I don't use PAM, systemctl status showed the auth failures). I changed the symlinks to point to the passwd.so files because that's what I use, but logs only show "afpd[]: AFP statistics: 0.51 KB read, 0.38 KB written"

How do I enable some extra logging to see what's happening or what should I look at? I can connect with the same username/password with SMB just fine.

**Comment by rdmark** [comment, marked as answer]
There are a few issues to unpack here:

First of all, if Netatalk was built with PAM, it means it found the PAM libraries on your system and tried to helpfully enable it. The safer way to disable PAM is to pass `-Dwith-pam=false` to Meson when you do setup.

Secondly, what kind of client are you trying to authenticate from? A fairly recent macOS system, or an old pre-OSX Mac? You may have to activate different UAMs or set different length password for older systems.

And to answer your question: You can turn on debug logging by adding this to the Global section in afp.conf: `log level = default:debug`

See the manual for more details:

- afp.conf options in https://netatalk.io/3.2/htmldocs/afp.conf.5
- authentication options in https://netatalk.io/3.2/htmldocs/configuration#authentication
